File: SDKs/shutterstock.com/1.1.5/typescript/src/sdk/models/operations/validate.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class ValidateQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=id" })
  id: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=tag" })
  tag?: string[];
}


export class ValidateHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=user-agent" })
  userAgent?: string;
}


export class ValidateRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: ValidateQueryParams;

  @SpeakeasyMetadata()
  headers: ValidateHeaders;
}


export class ValidateResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  testValidate?: shared.TestValidate;
}

File: SDKs/shutterstock.com/1.1.5/python/src/sdk/models/operations/validate.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from ..shared import testvalidate as shared_testvalidate


@dataclasses.dataclass
class ValidateQueryParams:
    id: int = dataclasses.field(metadata={'query_param': { 'field_name': 'id', 'style': 'form', 'explode': True }})
    tag: Optional[list[str]] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'tag', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class ValidateHeaders:
    user_agent: Optional[str] = dataclasses.field(default=None, metadata={'header': { 'field_name': 'user-agent', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class ValidateRequest:
    headers: ValidateHeaders = dataclasses.field()
    query_params: ValidateQueryParams = dataclasses.field()
    

@dataclasses.dataclass
class ValidateResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    test_validate: Optional[shared_testvalidate.TestValidate] = dataclasses.field(default=None)
    
